docs/component_spec.md

```diff
@@ -0,0 +1,173 @@
+# Cytoxnet Component Specification
+The package is intended as an exploration and a toolbox for the prediction of
+cytotoxicity of compounds, but aims to leverage datasets of other targets
+for transfer learning.
+***
+***
+## package format
+```
+/cytoxnet
+|-/dataprep
+| |-io.py
+| |-featurize.py
+| |-dataprep.py
+| |-analyze.py
+|-/models
+| |-models.py
+| |-opt.py
+| |-analyze.py
+|-/tests
+```
+***
+
+## `cytoxnet/dataprep/`
+
+### `cytoxnet/dataprep/io.py`
+
+```
+load_file(datafile: str,
+          cols: str or list of str = None) -> pandas DataFrame
+```
+- __Use Case__: (1)
+- __Inputs__: File location string, columns to keep
+- __Outputs__: Dataframe of dataset on file
+
+```
+load_XXX(prepare: bool = False)
+```
+- __Use Case__: (1)
+- __Inputs__: Whether or not to featurize and prepare set using a default
+  pipeline
+- __Outputs__: Dataframe of package dataset
+
+### `cytoxnet/dataprep/featurize.py`
+
+```
+clean_dataframe(dataframe,
+                cols: str or list of str = None,
+                duplicates: str = 'drop',
+                nans: str = 'drop') -> pandas DataFrame
```

Comment:
Will the nans argument take specific columns (ie using the `subset` argument of `pd.DataFrame.dropna` like we used for the data processing pipeline)?  Or will it just take a binary 'drop' or 'None'?

Reply:
it takes in options for handling nans eg 'drop' or 'zeros' or 'average' or any other way we might want to deal with missing values, and applies them to the columns passed to `col`